include/alpaka/kernel/TaskKernelOacc.hpp

```diff
@@ -44,301 +44,269 @@
 
 #if ALPAKA_DEBUG > ALPAKA_DEBUG_MINIMAL
-                std::cout << __func__
-                    << " blockSharedMemDynSizeBytes: " << blockSharedMemDynSizeBytes << " B" << std::endl;
+            std::cout << __func__
+                << " blockSharedMemDynSizeBytes: " << blockSharedMemDynSizeBytes << " B" << std::endl;
 #endif
-                // The number of blocks in the grid.
-                TIdx const gridBlockCount(gridBlockExtent.prod());
-                // The number of threads in a block.
-                TIdx const blockThreadCount(blockThreadExtent.prod());
+            // The number of blocks in the grid.
+            TIdx const gridBlockCount(gridBlockExtent.prod());
+            // The number of threads in a block.
+            TIdx const blockThreadCount(blockThreadExtent.prod());
 
-                if(gridBlockCount == 0 || blockThreadCount == 0)
-                { //! empty grid is a NOP
-                    return;
-                }
+            if(gridBlockCount == 0 || blockThreadCount == 0)
+            { //! empty grid is a NOP
+                return;
+            }
 
 #if ALPAKA_DEBUG >= ALPAKA_DEBUG_MINIMAL
-                std::cout << "threadElemCount=" << threadElemExtent[0u]
-                    << "\tgridBlockCount=" << gridBlockCount << std::endl;
+            std::cout << "threadElemCount=" << threadElemExtent[0u]
+                << "\tgridBlockCount=" << gridBlockCount << std::endl;
 #endif
-                auto argsD = m_args;
-                auto kernelFnObj = m_kernelFnObj;
-                dev.makeCurrent();
-                #pragma acc parallel num_workers(blockThreadCount) copyin(threadElemExtent,blockThreadExtent,argsD,gridBlockExtent) default(present)
+            auto argsD = m_args;
+            auto kernelFnObj = m_kernelFnObj;
+            dev.makeCurrent();
+            #pragma acc parallel num_workers(blockThreadCount) copyin(threadElemExtent,blockThreadExtent,argsD,gridBlockExtent) default(present)
+            {
                 {
+                    #pragma acc loop gang
+                    for(TIdx b = 0u; b<gridBlockCount; ++b)
                     {
-                        #pragma acc loop gang
-                        for(TIdx b = 0u; b<gridBlockCount; ++b)
-                        {
-                            ctx::CtxBlockOacc<TDim, TIdx> blockShared(
-                                gridBlockExtent,
-                                blockThreadExtent,
-                                threadElemExtent,
-                                b,
-                                blockSharedMemDynSizeBytes);
+                        CtxBlockOacc<TDim, TIdx> blockShared(
+                            gridBlockExtent,
+                            blockThreadExtent,
+                            threadElemExtent,
+                            b,
+                            static_cast<TIdx>(blockSharedMemDynSizeBytes)); // FIXME: added static_cast to make code compile, rethink about this
```

Comment:
I needed to add the cast to fix an error with GCC 10.2. Please fix this in a better way!